pkg/platform/packet/packet.go

```diff
@@ -451,6 +452,32 @@ func (c *config) checkReservationIDs() hcl.Diagnostics {
 	return diagnostics
 }
 
+// checkIPXEScriptURL checks that if os_version is specified then
```

Comment:
IIUC the focus here is `os_version`, not `ipxe_script_url`. Therefore I would name the function `validateOSVersion`.